synapse/storage/databases/main/user_directory.py

```diff
@@ -248,18 +248,15 @@ def _get_next_batch(
                     )
 
                 # Now update the room sharing tables to include this room.
-                to_insert = set()
                 is_public = await self.is_room_world_readable_or_publicly_joinable(
                     room_id
                 )
-                if is_public:
-                    for user_id in users_with_profile:
-                        to_insert.add(user_id)
-
-                    if to_insert:
-                        await self.add_users_in_public_rooms(room_id, to_insert)
-                        to_insert.clear()
+                if is_public and users_with_profile:
```

Comment:
although I think it ends up being the same functionally (because it's a no-op if `users_with_profile` is empty), the conditions under which we run the `else` branch (vis: either `is_public` is false, or `is_public` is true but `users_with_profile` is empty) make it confusing to read. Suggest:

```suggestion
                if is_public:
                  if users_with_profile:
```
(with suitable indentation fixes)

Reply:
Good spot, thank you---I hadn't thought of this at all!